python/pyspark/sql/connect/dataframe.py

```diff
@@ -692,10 +692,14 @@ def lateralJoin(
         on: Optional[Column] = None,
         how: Optional[str] = None,
     ) -> ParentDataFrame:
-        # TODO(SPARK-50134): Implement this method
-        raise PySparkNotImplementedError(
-            errorClass="NOT_IMPLEMENTED",
-            messageParameters={"feature": "lateralJoin()"},
+        self._check_same_session(other)
+        if how is not None and isinstance(how, str):
```

Comment:
What if `how` is not a string?

Reply:
It will fail with the `AnalysisException` above.